blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/DownstreamJobAction.java

```diff
@@ -0,0 +1,71 @@
+package io.jenkins.blueocean.service.embedded;
+
+import hudson.model.Action;
+import hudson.model.Run;
+
+import javax.annotation.CheckForNull;
+import java.util.Objects;
+
+/**
+ * Added to a job to provide details of downstream jobs triggered by it, as an inverse of Cause.UpstreamCause which
+ * is added to the downstream job during its creation.
+ */
+public class DownstreamJobAction implements Action {
+
+    private final String downstreamProject;
+    private final int downstreamBuild;
+
+    public DownstreamJobAction(Run<?, ?> run) {
+        this.downstreamProject = run.getParent().getName();
```

Comment:
I'm concerned this wont work with folders

Reply:
Any ideas on how I could verify this? How can I cook up a minimal folder project? I haven't messed with them.

Reply:
Yes a backend test is what you need. Look at existing ones. Does getName return the path? i thought it was getFullName

Reply:
You can use `Item.getFullName`, or more simply a single field with `Run.getExternalizableId`.